scripts/generator/Generator.js

```diff
@@ -0,0 +1,385 @@
+  /**
+   * Returns a hash containing the default value of the property.
+   * @param {Object} property - The component property.
+   * @return {Object} - A JSON object representing the property default value.
+   */
+  static defaultValue(property) {
+    const { type, defaultValue } = property;
+    const { name } = type;
+    const { value } = defaultValue || {};
+
+    if (value === undefined || value === 'undefined' || value === 'null') {
+      return {};
+    }
+
+    switch (name) {
+      case 'bool':
+        return value === 'true' ? { default: true } : {};
+      case 'number':
+        return { default: parseInt(defaultValue.value, 10) };
+      case 'string':
+        return { default: value.replace(/[']+/g, '') };
+      case 'enum':
+        return { default: value.replace(/[']+/g, '') };
+      default:
+        return {};
+    }
+  }
+
+  /**
+   * Creates a JSON representation of a property.
+   * @param {string} dependency - The dependency name.
+   * @param {string} prop - The property name.
+   * @param {Object} property - The component property.
+   * @return {Object|null} - A JSON object representing the property.
+   */
+  static createProperty(dependency, prop, property) {
+    const type = Generator.propertyType(dependency, prop, property);
+
+    if (type) {
+      return {
+        type,
+        display: Generator.humanize(prop),
+        ...Generator.defaultValue(property),
+        ...Generator.schema(dependency, prop, property),
+      };
+    }
+
+    return null;
+  }
+
+  /**
+   * Combines existing properties with new properties.
+   * @param {Object} config - The dependency configuration.
+   * @param {Object} props - All interpreted component properties.
+   * @param {Object} properties - Existing properties.
+   * @return {Object} - A hash containing the merged properties.
+   */
+  static mergeProperties(config, props, properties) {
+    const componentProps = { ...properties };
+    const { dependency, ignoredProperties = [] } = config;
+
+    // Generate new properties.
+    Object.keys(props).filter(key => properties[key] === undefined).forEach((prop) => {
+      if (ignoredProperties.indexOf(prop) > -1) {
+        return;
+      }
+
+      const property = Generator.createProperty(dependency, prop, props[prop]);
+
+      if (property) {
+        componentProps[prop] = property;
+      }
+    });
+
+    // Remove any properties that no longer exist.
+    Object.keys(properties).filter(key => props[key] === undefined).forEach((prop) => {
+      delete componentProps[prop];
+    });
+
+    return componentProps;
+  }
+
+  /**
+   * Creates a JSON representation of the dependency.
+   * @param {Object} config - The component configuration.
+   * @return {Object} - A JSON representing the dependency.
+   */
+  static createJSON(config) {
+    const { dependency, fileName, sourceFile } = config;
+    const { props = {} } = parse(fs.readFileSync(`node_modules/${dependency}/src/${sourceFile}.jsx`));
```

Comment:
These may be of interest to you:

- https://github.com/cerner/terra-core/issues/1795
- https://github.com/cerner/terra-core/pull/1791